tests/defaults.rs

```diff
@@ -0,0 +1,430 @@
+        .set_default(GooseDefault::Host, host.as_str())
+        .set_default(GooseDefault::Users, USERS)
+        .set_default(GooseDefault::RunTime, RUN_TIME)
+        .set_default(GooseDefault::HatchRate, HATCH_RATE)
+        .set_default(GooseDefault::LogLevel, LOG_LEVEL)
+        .set_default(GooseDefault::MetricsFile, metrics_file.as_str())
+        .set_default(GooseDefault::MetricsFormat, LOG_FORMAT)
+        .set_default(GooseDefault::DebugFile, debug_file.as_str())
+        .set_default(GooseDefault::DebugFormat, LOG_FORMAT)
+        .set_default(GooseDefault::ThrottleRequests, THROTTLE_REQUESTS)
+        .set_default(GooseDefault::StatusCodes, true)
+        .set_default(GooseDefault::OnlySummary, true)
+        .set_default(GooseDefault::NoTaskMetrics, true)
+        .set_default(GooseDefault::StickyFollow, true)
+        .execute()
+        .unwrap();
+
+    validate_test(goose_metrics, index, about, &[metrics_file], &[debug_file]);
+}
+
+#[test]
+/// Load test confirming that Goose respects CLI options.
+fn test_no_defaults() {
+    // Multiple tests run together, so set a unique name.
+    let metrics_file = "nodefaults-".to_string() + METRICS_FILE;
+    let debug_file = "nodefaults-".to_string() + DEBUG_FILE;
+
+    // Be sure there's no files left over from an earlier test.
+    cleanup_files(vec![&metrics_file, &debug_file]);
+
+    let server = MockServer::start();
+
+    let index = Mock::new()
+        .expect_method(GET)
+        .expect_path(INDEX_PATH)
+        .return_status(200)
+        .create_on(&server);
+    let about = Mock::new()
+        .expect_method(GET)
+        .expect_path(ABOUT_PATH)
+        .return_status(200)
+        .create_on(&server);
+
+    let mut config = common::build_configuration(&server);
+    config.no_metrics = false;
+    config.users = Some(USERS);
+    config.run_time = RUN_TIME.to_string();
+    config.hatch_rate = HATCH_RATE;
+    config.log_level = LOG_LEVEL as u8;
+    config.metrics_file = metrics_file.to_string();
+    config.metrics_format = LOG_FORMAT.to_string();
+    config.debug_file = debug_file.to_string();
+    config.debug_format = LOG_FORMAT.to_string();
+    config.throttle_requests = Some(THROTTLE_REQUESTS);
+    config.no_reset_metrics = true;
+    config.status_codes = true;
+    config.only_summary = true;
+    config.no_task_metrics = true;
+    config.sticky_follow = true;
+
+    let goose_metrics = crate::GooseAttack::initialize_with_config(config)
+        .setup()
+        .unwrap()
+        .register_taskset(taskset!("Index").register_task(task!(get_index)))
+        .register_taskset(taskset!("About").register_task(task!(get_about)))
+        .execute()
+        .unwrap();
+
+    validate_test(goose_metrics, index, about, &[metrics_file], &[debug_file]);
+}
+
+#[test]
+#[cfg_attr(not(feature = "gaggle"), ignore)]
+/// Load test confirming that Goose respects configured gaggle-related defaults.
+fn test_gaggle_defaults() {
```

Comment:
@LionsAd We have a test file named `gaggle`, and we have this test file named `defaults`. Should this test live here in `defaults`, or in `gaggle`? (It's a combination of both features).

Reply:
I'm thinking maybe I should just remove the `gaggle.rs` test file, and spread gaggle tests out into all the other tests, as in many cases functionality should be tested both in standalone mode, as well as in gaggle mode.

Reply:
I've created a followup issue for this: https://github.com/tag1consulting/goose/issues/158